core/ledger/pvtdatastorage/reconcile_missing_pvtdata.go

```diff
@@ -114,115 +200,144 @@ func (p *oldBlockDataProcessor) constructExpiryKeyFromDataEntry(dataEntry *dataE
 	}, nil
 }
 
-func (p *oldBlockDataProcessor) getExpiryDataFromEntriesOrStore(entries *entriesForPvtDataOfOldBlocks, expiryKey expiryKey) (*ExpiryData, error) {
-	if expiryData, ok := entries.expiryEntries[expiryKey]; ok {
+func (p *oldBlockDataProcessor) getExpiryDataFromEntriesOrStore(expKey expiryKey) (*ExpiryData, error) {
+	if expiryData, ok := p.entries.expiryEntries[expKey]; ok {
 		return expiryData, nil
 	}
 
-	expiryData, err := p.getExpiryDataOfExpiryKey(&expiryKey)
+	expData, err := p.db.Get(encodeExpiryKey(&expKey))
 	if err != nil {
 		return nil, err
 	}
-	return expiryData, nil
+	if expData == nil {
+		return nil, nil
+	}
+
+	return decodeExpiryValue(expData)
 }
 
-func (p *oldBlockDataProcessor) getMissingDataFromEntriesOrStore(entries *entriesForPvtDataOfOldBlocks, nsCollBlk nsCollBlk) (*bitset.BitSet, error) {
-	if missingData, ok := entries.missingDataEntries[nsCollBlk]; ok {
-		return missingData, nil
+func (p *oldBlockDataProcessor) getMissingDataFromEntriesOrStore(group elgMissingDataGroup, nsCollBlk nsCollBlk) (*bitset.BitSet, error) {
```

Comment:
I find the above a bit confusing because we need dbKey only if entries is not present. I would prefer to keep the current code as-is.

Reply:
Well, that is easy to solve by using `encodingFunc` instead of `dbKey` but the two interleaving flows within a function is not a normal style, if avoidable IMO.

Reply:
`encodingFunc` also looks like a hack to me. I will split this function into two as we are okay with certain amount of code duplication.

Reply:
Sure, I can compromise with this if you duplication. I'll prefer that over the current form.